File: LeetCode/Easy/389 - Find the Difference/389 - Find the Difference.js
/**
 * @param {string} s
 * @param {string} t
 * @return {character}
 */
 var findTheDifference = function(s, t) {
    const fr1 = countChars(s), fr2 = countChars(t);
    
    for (let i = 0; i < 26; ++i) {
        if (fr1[i] != fr2[i]) 
            return String.fromCharCode(i + 'a'.charCodeAt(0));
    }
};

var countChars = function(str) {
    const fr = Array(26).fill(0);
    for (const chr of str) fr[charCode(chr)]++;
    return fr;
};

var charCode = function(chr) {
    return chr.charCodeAt(0) - 'a'.charCodeAt(0);
};

File: LeetCode/Easy/389 - Find the Difference/389 - Find the Difference.cpp
class Solution {
public:
    char findTheDifference(string s, string t) {
        vector<int> fr(26);
        for (auto c : s) fr[c - 'a']++;
        for (auto c : t) fr[c - 'a']--;
        char ret;
        for (char c = 'a'; c <= 'z'; ++c) {
            if (fr[c - 'a']) {
                ret = c;
                break;
            }
        }
        return ret;
    }
};